packages/quantic/force-app/main/default/lwc/quanticFoldedResultList/quanticFoldedResultList.js

```diff
@@ -0,0 +1,161 @@
+   * @api
+   * @type {string}
+   * @defaultValue `'foldingparent'`
+   */
+  @api parentField = 'foldingparent';
+  /**
+   * The name of the field that uniquely identifies a result within collection.
+   * @api
+   * @type {string}
+   * @defaultValue `'foldingchild'`
+   */
+  @api childField = 'foldingchild';
+  /**
+   * The number of child results to fold under the root collection element before expansion.
+   * @api
+   * @type {number}
+   * @defaultValue `2`
+   */
+  @api numberOfFoldedResults = 2;
+
+  /** @type {AnyHeadless} */
+  headless;
+  /** @type {FoldedResultList} */
+  foldedResultList;
+  /** @type {ResultTemplatesManager} */
+  resultTemplatesManager;
+  /** @type {ResultsPerPage} */
+  resultsPerPage;
+  /** @type {FoldedResultListState} */
+  state;
+  /** @type {boolean} */
+  showPlaceholder = true;
+  /** @type {number} */
+  numberOfResults;
+  /** @type {Function} */
+  unsubscribe;
+  /** @type {Function} */
+  unsubscribeResultsPerPage;
+
+  connectedCallback() {
+    registerComponentForInit(this, this.engineId);
+  }
+
+  renderedCallback() {
+    initializeWithHeadless(this, this.engineId, this.initialize);
+  }
+
+  /**
+   * @param {SearchEngine} engine
+   */
+  initialize = (engine) => {
+    this.headless = getHeadlessBundle(this.engineId);
+    this.foldedResultList = this.headless.buildFoldedResultList(engine, {
+      options: {
+        fieldsToInclude: this.fields,
+        folding: {
+          numberOfFoldedResults: this.numberOfFoldedResults,
+          childField: this.childField,
+          collectionField: this.collectionField,
+          parentField: this.parentField,
+        },
+      },
+    });
+    this.unsubscribe = this.foldedResultList.subscribe(() =>
+      this.updateState()
+    );
+
+    this.resultsPerPage = this.headless.buildResultsPerPage(engine);
+    this.unsubscribeResultsPerPage = this.resultsPerPage.subscribe(() =>
+      this.updateResultPerPage()
+    );
+
+    this.resultTemplatesManager =
+      this.headless.buildResultTemplatesManager(engine);
+    this.registerTemplates();
+  };
+
+  registerTemplates() {
+    this.dispatchEvent(
+      new CustomEvent('registerresulttemplates', {
+        bubbles: true,
+        detail: this.resultTemplatesManager,
+      })
+    );
+  }
+
+  disconnectedCallback() {
+    this.unsubscribe?.();
+    this.unsubscribeResultsPerPage?.();
+  }
+
+  updateState() {
+    this.state = this.foldedResultList.state;
+    this.showPlaceholder =
+      this?.state?.isLoading &&
+      !this?.state?.hasError &&
+      !this?.state?.firstSearchExecuted &&
+      !this?.state?.hasResults;
+  }
+
+  updateResultPerPage() {
+    this.numberOfResults = this.resultsPerPage.state.numberOfResults;
+  }
+
+  get fields() {
+    if (this.fieldsToInclude.trim() === '') return [];
+    return this.fieldsToInclude
+      .split(',')
+      .map((field) => field.trim())
+      .filter((field) => field.length > 0);
+  }
```

Comment:
The advantage of filtering the empty fields at the end is that the separate `if` is not necessary anymore.

```suggestion
  get fields() {
    return this.fieldsToInclude
      .split(',')
      .map((field) => field.trim())
      .filter((field) => field.length > 0);
  }
```